lib/samples/create_and_edit_geometries/create_and_edit_geometries.dart

```diff
@@ -0,0 +1,607 @@
+      children: [
+        // A drop down button for selecting geometry type.
+        DropdownButton(
+          alignment: Alignment.center,
+          hint: const Text(
+            'Geometry Type',
+            style: TextStyle(
+              color: Colors.deepPurple,
+            ),
+          ),
+          icon: const Icon(Icons.arrow_drop_down),
+          iconEnabledColor: Colors.deepPurple,
+          iconDisabledColor: Colors.grey,
+          style: const TextStyle(color: Colors.deepPurple),
+          value: _selectedGeometryType,
+          items: configureGeometryTypeMenuItems(),
+          // If the geometry editor is already started then we don't enable editing with another geometry type.
+          onChanged: !_geometryEditorIsStarted
+              ? (geometryType) =>
+                  startEditingWithGeometryType(geometryType as GeometryType)
+              : null,
+        ),
+        // A drop down button for selecting a tool.
+        DropdownButton(
+          alignment: Alignment.center,
+          hint: const Text(
+            'Tool',
+            style: TextStyle(color: Colors.deepPurple),
+          ),
+          iconEnabledColor: Colors.deepPurple,
+          style: const TextStyle(color: Colors.deepPurple),
+          value: _selectedTool,
+          items: configureToolMenuItems(),
+          onChanged: (tool) {
+            if (tool != null) {
+              setState(() => _selectedTool = tool);
+              _geometryEditor.tool = tool;
+            }
+          },
+        ),
+        // A button to toggle the visibility of the editing toolbar.
+        IconButton(
+          onPressed: () => setState(() => _showEditToolbar = !_showEditToolbar),
+          icon: const Icon(Icons.edit, color: Colors.deepPurple),
+        ),
+      ],
+    );
+  }
+
+  Widget buildEditingToolbar() {
+    // A toolbar of buttons with icons for editing functions. Tooltips are used to aid the user experience.
+    return Padding(
+      padding: const EdgeInsets.only(bottom: 100, right: 5),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          Column(
+            mainAxisSize: MainAxisSize.max,
+            mainAxisAlignment: MainAxisAlignment.end,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              Row(
+                children: [
+                  // A button to call undo on the geometry editor, if enabled.
+                  Tooltip(
+                    message: 'Undo',
+                    child: ElevatedButton(
+                      style: _buttonStyle,
+                      onPressed:
+                          _geometryEditorIsStarted && _geometryEditorCanUndo
+                              ? () => _geometryEditor.undo()
+                              : null,
+                      child: const Icon(Icons.undo),
+                    ),
+                  ),
+                  const Padding(padding: EdgeInsets.all(2)),
```

Comment:
I'm not sure what this is doing. What is a Padding without a child to apply padding to? I think we'd usually use SizedBox here?

Reply:
Yeah, it's just a widget adding spacing. I usually use Padding but happy to switch to SizedBox if that's preferred.

Reply:
Yeah the padding adds space around its child widget. Without a child widget it's padding around a nothing dot in the center. Which means `EdgeInsets.all(2)` creates 4 pixels of horizontal space, and 4 pixels of vertical space that has no effect. So `SizedBox(width: 4.0)` is more to the point.